Pretty basic compile fail of GUI -  candidate expects 2 arguments, 4 provided

Linux Mint 18.3

Fresh:    git clone https://github.com/monero-project/monero-gui.git

./build.sh 

gives



/translations/monero-core_zh-tw.qm
Updating '/root/monero-gui/build/translations/monero-core_zh-tw.qm'...
Removing translations equal to source text in '/root/monero-gui/build/translations/monero-core_zh-tw.qm'...
    Generated 369 translation(s) (369 finished and 0 unfinished)
    Ignored 104 untranslated source text(s)
g++ -c -m64 -pipe -fPIC -fstack-protector -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -Wformat -Wformat-security -fstack-protector -fstack-protector-strong -O2 -std=c++0x -Wall -W -D_REENTRANT -fPIC -DQT_NO_DEBUG -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../../monero-gui -I. -I../monero/include -I../src/libwalletqt -I../src/QR-Code-generator -I../src -I../monero/src -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtQuick -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtQml -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -I. -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++-64 -o main.o ../main.cpp
../main.cpp: In function ‘int main(int, char**)’:
../main.cpp:126:91: error: no matching function for call to ‘Monero::Wallet::init(char*&, const char [18], const char*, bool)’
 allet::init(argv[0], "monero-wallet-gui", logPath.toStdString().c_str(), true);
                                                                              ^
In file included from ../src/libwalletqt/WalletManager.h:6:0,
                 from ../main.cpp:42:
../monero/include/wallet/api/wallet2_api.h:440:18: note: candidate: virtual bool Monero::Wallet::init(const string&, uint64_t, const string&, const string&, bool, bool)
     virtual bool init(const std::string &daemon_address, uint64_t upper_transac
                  ^
../monero/include/wallet/api/wallet2_api.h:440:18: note:   no known conversion for argument 4 from ‘bool’ to ‘const string& {aka const std::__cxx11::basic_string<char>&}’
../monero/include/wallet/api/wallet2_api.h:559:17: note: candidate: static void Monero::Wallet::init(const char*, const char*)
     static void init(const char *argv0, const char *default_log_base_name);
                 ^
../monero/include/wallet/api/wallet2_api.h:559:17: note:   candidate expects 2 arguments, 4 provided
Makefile:986: recipe for target 'main.o' failed
make: *** [main.o] Error 1

--- a/README.md
+++ b/README.md
@@ -69,11 +69,11 @@ Packaging for your favorite distribution would be a welcome contribution!
 
 ### On Linux:
 
-(Tested on Ubuntu 16.04 x86, 16.10 x64, Gentoo x64 and Linux Mint 18 "Sarah" - Cinnamon x64)
+(Tested on Ubuntu 17.10 x64, Ubuntu 18.04 x64 and Gentoo x64)
 
 1. Install Monero dependencies
 
-  - For Ubuntu and Mint
+  - For Debian distributions (Debian, Ubuntu, Mint, Tails...)
 
 	`sudo apt install build-essential cmake libboost-all-dev miniupnpc libunbound-dev graphviz doxygen libunwind8-dev pkg-config libssl-dev libzmq3-dev`
 
@@ -81,27 +81,13 @@ Packaging for your favorite distribution would be a welcome contribution!
 
 	`sudo emerge app-arch/xz-utils app-doc/doxygen dev-cpp/gtest dev-libs/boost dev-libs/expat dev-libs/openssl dev-util/cmake media-gfx/graphviz net-dns/unbound net-libs/ldns net-libs/miniupnpc net-libs/zeromq sys-libs/libunwind`
 
-2. Grab an up-to-date copy of the monero-gui repository
+2. Install Qt:
 
-	`git clone https://github.com/monero-project/monero-gui.git`
+  *Note*: Qt 5.7 is the minimum version required to build the GUI. This makes **some** distributions (mostly based on debian, like Ubuntu 16.x or Linux Mint 18.x) obsolete. You can still build the GUI if you install an [official Qt release](https://wiki.qt.io/Install_Qt_5_on_Ubuntu), but this is not officially supported.
 
-3. Go into the repository
+  - For Ubuntu 17.10+
 
-	`cd monero-gui`
-
-4. Install the GUI dependencies
-
-  - For Ubuntu 16.04 x86
-
-	`sudo apt install qtbase5-dev qt5-default qtdeclarative5-dev qml-module-qtquick-controls qml-module-qtquick-controls2 qml-module-qt-labs-folderlistmodel qml-module-qtquick-xmllistmodel qttools5-dev-tools qml-module-qtquick-dialogs`
-
-  - For Ubuntu 16.04+ x64
-
-    `sudo apt install qtbase5-dev qt5-default qtdeclarative5-dev qml-module-qtquick-controls qml-module-qtquick-xmllistmodel qttools5-dev-tools qml-module-qtquick-dialogs qml-module-qt-labs-settings libqt5qml-graphicaleffects`
-
-  - For Linux Mint 18 "Sarah" - Cinnamon x64
-
-    `sudo apt install qml-module-qt-labs-settings qml-module-qtgraphicaleffects`
+    `sudo apt install qtbase5-dev qt5-default qtdeclarative5-dev qml-module-qtquick-controls qml-module-qtquick-controls2 qml-module-qtquick-dialogs qml-module-qtquick-xmllistmodel qml-module-qt-labs-settings qml-module-qt-labs-folderlistmodel qttools5-dev-tools`
 
   - For Gentoo
 
@@ -109,7 +95,7 @@ Packaging for your favorite distribution would be a welcome contribution!
 
   - Optional : To build the flag `WITH_SCANNER`
 
-    - For Ubuntu and Mint
+    - For Ubuntu
 
       `sudo apt install qtmultimedia5-dev qml-module-qtmultimedia libzbar-dev`
 
@@ -119,15 +105,17 @@ Packaging for your favorite distribution would be a welcome contribution!
 
       `emerge dev-qt/qtmultimedia:5 media-gfx/zbar`
 
-5. Build the GUI
 
-  - For Ubuntu and Mint
+3. Clone repository
 
-	`./build.sh`
+    `git clone https://github.com/monero-project/monero-gui.git`
 
-  - For Gentoo
+4. Build
 
-    `QT_SELECT=5 ./build.sh`
+    ```
+    cd monero-gui
+    ./build.sh
+    ```
 
 The executable can be found in the build/release/bin folder.
 

